src/test/java/org/minions/devfund/noemiguzman/SquarelotronTest.java

```diff
@@ -0,0 +1,220 @@
+        final int[][] expected = {{16, 15, 14, 13}, {12, 11, 10, 9}, {8, 7, 6, 5}, {4, 3, 2, 1}};
+        assertArrayEquals(expected, s4.getSquarelotron());
+    }
+    /**
+     * Method to test MainDiagonalFlip Dimension Two.
+     */
+    @Test
+    public void testMainDiagonalFlipDimensionTwo() {
+        s4 = new Squarelotron(SIZE4);
+        final int ring2 = 2;
+        final int[][] expected = {{1, 2, 3, 4}, {5, 6, 10, 8}, {9, 7, 11, 12}, {13, 14, 15, 16}};
+        Squarelotron temp5 = s4.mainDiagonalFlip(ring2);
+        assertArrayEquals(expected, temp5.getSquarelotron());
+    }
+    /**
+     * Method to test MainDiagonalFlip Dimension FiveRingThree.
+     */
+    @Test
+    public void testMainDiagonalFlipDimensionFiveRingThree() {
+        s4 = new Squarelotron(SIZE5);
+        final int ring3 = 3;
+        final int[][] expected = {{1, 2, 3, 4, 5}, {6, 7, 8, 9, 10}, {11, 12, 13, 14, 15}, {16, 17, 18, 19, 20},
+                {21, 22, 23, 24, 25}};
+        Squarelotron temp5 = s4.mainDiagonalFlip(ring3);
+        assertArrayEquals(expected, temp5.getSquarelotron());
+    }
+    /**
+     * Method to test UpsideDownFlip dimension four.
+     */
+    @Test
+    public void testUpsideDownFlipDimensionFour() {
+        s4 = new Squarelotron(SIZE4);
+        final int ring2 = 2;
+        // 4 by 4, but flip inner ring
+        final int[][] test4 = {{1, 2, 3, 4}, {5, 10, 11, 8}, {9, 6, 7, 12}, {13, 14, 15, 16}};
+        Squarelotron temp4 = s4.upsideDownFlip(ring2);
+        assertArrayEquals(test4, temp4.getSquarelotron());
+        // 4 by 4, but flip outer ring
+        final int[][] test5 = {{13, 14, 15, 16}, {9, 6, 7, 12}, {5, 10, 11, 8}, {1, 2, 3, 4}};
+        Squarelotron temp5 = s4.upsideDownFlip(1);
+        assertArrayEquals(test5, temp5.getSquarelotron());
+    }
+    /**
+     * Method to test UpsideDownFlip dimension one new object.
+     */
+    @Test
+    public void testUpsideDownFlipDimensionOneNewObject() {
+        s = new Squarelotron(1);
+        Squarelotron temp = s.upsideDownFlip(1);
+        final int[][] test = {{1}};
+        assertArrayEquals(test, temp.getSquarelotron());
+    }
+    /**
+     * Method to test UpsideDownFlip new object.
+     */
+    @Test
+    public void testUpsideDownFlipNewObject() {
+        s = new Squarelotron(SIZE3);
+        Squarelotron temp = s.upsideDownFlip(1);
+        final int[][] test = {{7, 8, 9}, {4, 5, 6}, {1, 2, 3}};
+        assertArrayEquals(test, temp.getSquarelotron());
+    }
+    /**
+     * Method to test UpsideDownFlip dimension three.
+     */
+    @Test
+    public void testUpsideDownFlipDimensionThree() {
+        s = new Squarelotron(SIZE3);
+        Squarelotron temp = s.upsideDownFlip(1);
+        final int[][] test = {{7, 8, 9}, {4, 5, 6}, {1, 2, 3}};
+        assertArrayEquals(test, temp.getSquarelotron());
+    }
+    /**
+     * Method to test UpsideDownFlip dimension one.
+     */
+    @Test
+    public void testUpsideDownFlipDimensionOne() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "testUpsideDownFlipDimensionOneNewObject" on line 152. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)